packages/astro/src/react/controlComponents.tsx

```diff
@@ -21,6 +22,22 @@ export function SignedIn(props: PropsWithChildren) {
   return props.children;
 }
 
+export const ClerkLoaded = ({ children }: React.PropsWithChildren<unknown>): JSX.Element | null => {
+  const { isLoaded } = useStore($csrState);
+  if (!isLoaded) {
+    return null;
+  }
+  return <>{children}</>;
+};
+
+export const ClerkLoading = ({ children }: React.PropsWithChildren<unknown>): JSX.Element | null => {
+  const { isLoaded } = useStore($csrState);
+  if (isLoaded) {
+    return null;
+  }
+  return <>{children}</>;
+};
+
```

Comment:
Seems like we are making use of the wrong `useStore` hook here. That one is using authAsyncStorage during SSR which may cause issues during hydration as i am not expecting `$csrState` and `authAsyncStorage.getStore()` to match. Also `isLoaded` from useAuth() is not an indication that clerk-js has loaded, but an indication that `useAuth()` can be safely accessed. In a SSR app `isLoaded` will always be true even when clerk-js has not loaded yet.

Reply:
Thanks for explaining this again. Also I realized that we dont really need the features of this custom `useStore` since clerk loading and loaded components expects to be working on client only.

I'm exporting the `useStore` have in the react utils file and use inside control components, which will also be used by other control components if we decide to add more